python/lsst/daf/persistence/policy.py

```diff
@@ -0,0 +1,392 @@
+    That is, foo.getValue('a.b.c') is the same as foo['a']['b']['c'] is the same as foo['a.b.c'], and either
+    of these syntaxes may be used.
+
+    Storage formats supported:
+    - yaml: read and write is supported.
+    - pex policy: read is supported, although this is deprecated and will at some point be removed.
+    """
+
+    defaultPolicyInitData = collections.namedtuple('defaultPolicyInitData', 'productName filePath repos')
+    # Contains data for loading a default policy from a package:
+    # product name: the name of the pacage
+    # filePath: the name of the policy file
+    # the file path from the root of the directory to the dir that contains the file.
+
+    def __init__(self, other=None):
+        """Constructor. Accepts an optional initializer var which may be one of:
+        - a path to a yaml or paf configuration file
+        - an instance of a Policy
+        - an instance of pex_policy
+        - default policy data with a populated instance of a defaultPolicyInitData
+            Policy.defaultPolicyInitData('daf_persistence', 'testPolicy.yaml', 'tests')
+
+        :param other: data to be loaded
+        :return:
+        """
+        UserDict.UserDict.__init__(self)
+        self.__init(other)
+
+    def __init(self, other=None):
+        if isinstance(other, pexPolicy.Policy):
+            self.__initFromPexPolicy(other)
+        elif isinstance(other, basestring): # (could create a named tuple for this)
+            self.__initFromFile(other)
+        elif isinstance(other, collections.Mapping) or isinstance(other, Policy):
+            # don't assign other to self.data directly; need assign via parse in self.update.
+            self.update(other)
+        elif isinstance(other, Policy.defaultPolicyInitData):
+            self.__initFromDefaultPolicy(*other)
+
+    def __initFromFile(self, paths):
+        """Load a file from path. If path is a list, will pick one to use, according to order specified
+        by extensionPreference.
+
+        :param path: string or list of strings, to a persisted policy file.
+        :param extensionPreference: the order in which to try to open files. Will use the first one that
+        succeeds.
+        :return:
+        """
+        extensionPreference = ('yaml', 'paf')
+        matchedPath = None
+        if isinstance(paths, basestring):
+            paths = [paths]
+        for ext in extensionPreference:
+            if matchedPath is not None:
+                break
+            for path in paths:
+                if path.endswith(ext):
+                    matchedPath = path
+                    break
+        policy = None
+        if matchedPath is None:
+            policy = Policy() # return a default policy? throw? WTD?
+        elif matchedPath.endswith('yaml'):
+            self.__initFromYamlFile(matchedPath)
+        elif matchedPath.endswith('paf'):
+            policy = pexPolicy.Policy_createPolicy(matchedPath)
```

Comment:
You should never have to use {class}_{name} SWIG-wrapped methods.  You should use `pexPolicy.Policy.createPolicy()` here, I think.